docs/Spark/configuration/configuration.md

```diff
@@ -30,21 +31,56 @@ Hadoop configurations as name-value pairs. The name-value pairs will be set insi
 
 Config values which can be set at Pipeline level and then be accessed inside any component in the Pipeline. [Multiple instances](#pipeline-configuration-instances)
 of configuration can be created per Pipeline.
+
 ![Configurations - Common](img/config-pipeline-eg1.png)
 
----
+### Syntax for using configuration inside Gems
+
+We support language-specific config syntaxes for different data types and coding languages for configurations inside of your Spark Gems. We also support a common Jinja config syntax. You can use either syntax or a combination of them.
 
-Syntax for using configuration inside Gems:
+#### Language-specific Config syntax
+
+See the following language-specific config syntax for SQL, Scala, and Python in Spark.
 
 ![Visual Language Selection](img/config-pipeline-visual-language.png)
 
-For visual language SQL : `'$config_name'`
+- For visual language SQL: `'$config_name'`
+
+- For visual language Scala/Python: `Config.config_name`
+
+- For using Spark expression with visual language SQL: `expr('$config_name')`
+
+- For using Spark expression with visual language Scala/Python: `expr(Config.config_name)`
+
+#### Jinja Config syntax
+
+You can choose to use a common Jinja config syntax for configurations inside of your Spark Gems.
+
+You must enable it by navigating to **...** > **Pipeline Settings**.
+
+![Pipeline Settings](img/navigate-pipeline-settings.png)
+
+Under the **Code** section, click to enable Jinja config syntax. This setting is toggled on by default for new Pipelines.
+
+![Enable Jinja Config syntax](img/enable-jinja-config-syntax.png)
 
-For visual language Scala/Python : `Config.config_name`
+See the following syntax for Jinja in Spark.
 
-For using Spark expression with visual language SQL : `expr('$config_name')`
+- For Jinja: `{{config_name}}`
 
-For using Spark expression with visual language Scala/Python : `expr(Config.config_name)`
+- For using Spark expression with Jinja: `expr('{{config_name}}')`
+
+So for example, instead of using `'$read_env'` for SQL or `Config.read_env` for Scala/Python, you can use `{{read_env}}`.
+
+Jinja config syntax supports the following functionalities:
+
+- **Concatenation** - You can define multiple Jinja syntaxes in the same string, and the generated code will be a formatted string. For example, `'{{c_string}}___{{ c_int }}'`
```

Comment:
We can also mention that to use datatype as it is, like use an integer field as an integer. Use it as `{{ integer_field }}`. Doing something like `<some_character>{{ integer_field }}<some_character>` will be treated as a string instead.

Reply:
Done.

Reply:
Thank you for the comments.